packages/wdio-allure-reporter/src/index.ts

```diff
@@ -1,605 +1,686 @@
+        process.on(events.addSeverity, this.addSeverity.bind(this))
+        process.on(events.addIssue, this.addIssue.bind(this))
+        process.on(events.addTestId, this.addTestId.bind(this))
+        process.on(events.addEnvironment, this.addEnvironment.bind(this))
+        process.on(events.addAttachment, this.addAttachment.bind(this))
+        process.on(events.addDescription, this.addDescription.bind(this))
+        process.on(events.startStep, this.startStep.bind(this))
+        process.on(events.endStep, this.endStep.bind(this))
+        process.on(events.addStep, this.addStep.bind(this))
+        process.on(events.addArgument, this.addArgument.bind(this))
+    }
+
+    onRunnerStart(runner: any) {
+        this.config = runner.config
+        this.capabilities = runner.capabilities
+        this.isMultiremote = runner.isMultiremote || false
+    }
+
+    onSuiteStart(suite: SuiteStats) {
+        if (this.options.useCucumberStepReporter) {
+            if (suite.type === 'feature') {
+                // handle cucumber features as allure "suite"
+                return this.allure.startSuite(suite.title)
+            }
+
+            // handle cucumber scenarii as allure "case" instead of "suite"
+            this.allure.startCase(suite.title)
+            const currentTest = this.allure.getCurrentTest()
+            this.getLabels(suite).forEach(({ name, value }) => {
+                currentTest.addLabel(name, value)
+            })
+            if (suite.description) {
+                this.addDescription(suite)
+            }
+            return this.setCaseParameters(suite.cid)
+        }
+
+        const currentSuite = this.allure.getCurrentSuite()
+        const prefix = currentSuite ? currentSuite.name + ': ' : ''
+        this.allure.startSuite(prefix + suite.title)
+    }
+
+    onSuiteEnd(suite: SuiteStats) {
+        if (this.options.useCucumberStepReporter && suite.type === 'scenario') {
+            // passing hooks are missing the 'state' property
+            suite.hooks = suite.hooks!.map((hook) => {
+                hook.state = hook.state ? hook.state : 'passed'
+                return hook
+            })
+            const suiteChildren = [...suite.tests!, ...suite.hooks]
+            const isPassed = !suiteChildren.some(item => item.state !== 'passed')
+            if (isPassed) {
+                return this.allure.endCase('passed')
+            }
+
+            // A scenario is it skipped if is not passed and every steps/hooks are passed or skipped
+            const isSkipped = suiteChildren.every(item => [PASSED, SKIPPED].indexOf(item.state!) >= 0)
+            if (isSkipped) {
+                return this.allure.endCase(PENDING)
+            }
+
+            // Only close passing and skipped tests because
+            // failing tests are closed in onTestFailed event
+            return
+        }
+
+        this.allure.endSuite()
+    }
+
+    onTestStart(test: TestStats | HookStats) {

```

Comment:
```suggestion
    onTestStart(test: TestStats) {

```

Reply:
Our code currently use this method for both tests and hooks, so we need to override type here

Reply:
Can you point out where we do that?